cirq-core/cirq/testing/routing_devices.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Provides test devices that can validate circuits during a routing procedure."""
+
+from typing import Optional, TYPE_CHECKING
+import networkx as nx
+
+from cirq import devices
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class RoutingTestingDevice(devices.Device):
+    """Testing device to be used only for testing qubit connectivity in routing procedures."""
```

Comment:
This seems like a pretty general-purpose `Device` where it just takes a metadata and validates as expected. Do we need our own subclass for it? Why can't we use `Device` or `GridDevice` directly?

Reply:
`Device` is an abstract class and `GridDevice` is in cirq_google. From what I understand, we can't have anything in cirq_core depend on cirq_google.

Reply:
This is why we have to create our own device for testing purposes and do the crate the grid graph manually (potentially with some help from networkx's `grid_2d_graph()` function or another). Another option I guess would be to user `GridDeviceMetadata` which for some reason is in cirq-core. The only issue with that is that is requires a list of qubit-pairs to initialize it, which would essentially result in recreating the grid graph on our own too.

Reply:
Should we change the constructor of this device to expect only a graph and then create unrelated abstractions, like device metadata, as part of the constructor of this device?